internal/sourcecode/templates/lucky/Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM crystallang/crystal:1.3.2-alpine as crystal_dependencies
+ENV LUCKY_ENV=production
+ENV SKIP_LUCKY_TASK_PRECOMPILATION=1
+WORKDIR /shards
+COPY shard.* ./
+RUN  shards install --production
+
+FROM node:alpine as asset_build
+WORKDIR /assets
+COPY . .
+RUN yarn install
+RUN yarn prod
+
+FROM crystallang/crystal:1.3.2-alpine as lucky_tasks_build
+ENV LUCKY_ENV=production
+RUN apk --no-cache add yaml-static
+COPY . .
+COPY --from=crystal_dependencies /shards/lib lib
+COPY --from=asset_build /assets/public public
+RUN crystal build --static --release tasks.cr -o /usr/local/bin/lucky
+
+FROM crystallang/crystal:1.3.2-alpine as lucky_webserver_build
+WORKDIR /webserver_build
+RUN apk --no-cache add yaml-static coreutils
+ENV LUCKY_ENV=production
+COPY . .
+COPY --from=crystal_dependencies /shards/lib lib
+COPY --from=asset_build /assets/public public
+RUN shards build --production --static --release
+RUN mv ./bin/`ls ./bin/ | head -1` /usr/local/bin/webserver
```

Comment:
I'm curious about this line - could you add a comment explaining why it's needed? Would a symlink work? Could Lucky create a generic `bin/webserver`?

Reply:
```suggestion
# shard.yml targets can be named anything, select the bin/ file outputted by build
RUN mv ./bin/`ls ./bin/ | head -1` /usr/local/bin/webserver
```

Reply:
When crystal builds, it's based off of the user's constant name by default.
Is there a way to template the Dockerfile built in? We could read the constant name from shard.yml then inject it into the dockerfile

Reply:
We don't want to get into templating Dockerfiles again - there lies madness! I was suggesting that lucky drop a generic binary or symlink instead of using a custom name. This is something that came up with Phoenix. They made it simpler by creating a `bin/webserver` that works out of the box.